src/geom/length.rs

```diff
@@ -34,6 +36,20 @@ impl Length {
             None
         }
     }
+
+    /// Get a field from this length.
+    pub fn at(&self, field: &str) -> StrResult<Value> {
+        let round_four_digits = |n| ((n as f64) * 1e4).round() / 1e4;
```

Comment:
why this?

Reply:
This was because floating point imprecision was causing issues - 3cm wouldn't be equal to `(3cm).cm` because the latter would return something like `3.000000004`. However, I think it would be better if we turned `.cm` and other conversions into methods where you can specify the precision.